internal/exec/valdiate.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2017 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exec
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/coreos/ignition/config"
+)
+
+func Validate(filename string) error {
+	b, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+
+	_, r, err := config.Parse(b)
+	fmt.Printf("Validation report:\n%v\n", r)
+	if err != nil {
+		return err
```

Comment:
Sometimes this prints things and sometimes it returns errors.  The caller prints to stderr, but this function prints to stdout.  I'd suggest putting all of the printing in one place.

Reply:
I'll move the printing to main, make `exec.Validate` return a `report` and `error`. 

As for `stdout`/`stderr`: It prints to stderr in the case that it cannot validate (not an Ignition config, file not found, etc). In this case an invalid config that was parsed correctly, Ignition should print it's report to `stdout`, since Ignition successfully generated the report. In the cases that it failed to valdiate it should print to `stderr` since Ignition failed to validate it.